pithos/pithos.py

```diff
@@ -1460,22 +1463,63 @@ def restore_position(self):
         if not x is None and not y is None:
             self.move(int(x), int(y))
 
+    def restore_size(self):
+        """ resizes window to size stored in preferences """
+        w, h = self.settings['win-size']
+        if w is not None and h is not None:
+            self.resize(int(w), int(h))
+
+    def restore_maximized(self):
+        if self.settings['win-maxed']:
+            self.maximize()
+
+    def restore_window(self):
+        # Getting the window position
+        # does not work in Wayland.
+        if self.running_in_x:
+            self.restore_position()
```

Comment:
I believe it is just ignored on Wayland so there is no harm in always calling it?

Reply:
It's not ignored, It returns a value, just not the correct value, if I remember for me it was always returning 15, 23 no matter what.

Reply:
Strange but alright.

Reply:
You still get configure events so it's set to the invalid value. The end result is that if you switch from an X session to a Wayland session and back again to an X session it basically wipes your setting. Which isn't a huge deal. Remembering this stuff is a waste of time anyway... But generally we could save a tiny bit of resources in Wayland by just disconnecting the handler altogether. At the very least we should block the handler when we restore the position.